docs/setup.md

```diff
@@ -0,0 +1,702 @@
+    using GCE.)
+0.  If your machine has less than 2GB of memory (check with `free -h`;
+    the very first number shown is total RAM), you will need to create
+    a swap file (this is mandatory on `f1-micro` instances):
+    ```
+    sudo -i
+    fallocate -l 2G /swapfile
+    chmod 600 /swapfile
+    mkswap /swapfile
+    swapon /swapfile
+    swapon -s
+    sh -c 'echo "/swapfile none swap sw 0 0" >> /etc/fstab'
+    exit
+    ```
+0.  Check for and install system updates, then install
+    [nginx](https://en.wikipedia.org/wiki/Nginx) and
+    [git](https://git-scm.com/):
+    ```
+    sudo apt-get update
+    sudo apt-get upgrade –y
+    sudo apt-get install -y nginx git
+    ```
+0.  Optionally install a text editor of your choice.  Debian comes
+    with [`vim`](https://www.vim.org/) and
+    [`nano`](https://www.nano-editor.org/) preinstalled; Emacs users
+    might feel more at home with the lightweight editors
+    [`mg`](https://github.com/hboetes/mg),
+    [`jove`](https://github.com/jonmacs/jove) or
+    [`zile`](https://www.gnu.org/software/zile/), or opt for
+    `emacs-nox` which is GNU Emacs without X Windows bindings.
+    ```
+    sudo apt-get install mg
+    ```
+0.  Install [node.js](https://nodejs.org/).  Code City depends on
+    version 12, which is more recent than the version included in
+    Debian 10, so we will obtain it via [NodeSource](
+    https://github.com/nodesource/distributions):
+    ```
+    sudo -i
+    curl -sL https://deb.nodesource.com/setup_12.x | bash -
+    apt-get install -y nodejs
+    exit
+    ```
+0.  Verify the correct version of node is installed:
+    ```
+    $ node –-version v12.18.2
+    ```
+    (Actual version may be later than 12.18.)
+
+### Get TLS Certificates
+
+In order to allow incoming HTTPS connections, you will need an
+[TLS](https://en.wikipedia.org/wiki/Transport_Layer_Security) [server
+certificate](
+https://en.wikipedia.org/wiki/Public_key_certificate#TLS/SSL_server_certificate
+).  There are two types:
+
+*   An ordinary certificate covers one or more specific domain names,
+    like `www.example.org`.
+*   A wildcard certificate includes one or more wildcard domains, like
+    `*.example.org`.  If you’re using a wildcard DNS entry, a
+
+You will need to get a TLS certificate covering the [set of DNS entries
+you [created earlier](#set-up-an-ip-address-and-domain-name).  If (as
+recommended) you created a wildcard DNS entry, you will also need a
+corresponding wildcard TLS certificate.
+
+There are various ways to get a TLS certificate, but a free and easy
+way is to use [Certbot](https://certbot.eff.org/) to get one from
+[Let’s Encrypt](https://letsencrypt.org/).  That’s what we’ll do here.
+
+#### Getting a wildcard certificate
+
+To use Certbot to get a wildcard certificate, you will need to use the
+[`dns-01` challenge](
+https://letsencrypt.org/docs/challenge-types/#dns-01-challenge), which
+requires being able to create [DNS TXT records](
```

Comment:
Can I recommend using [callout-links](https://www.google.com/search?q=external+link+icon) for what are essentially [dictionary](https://www.oed.com/) definitions.  This reserves actual [links](https://www.directindustry.com/industrial-manufacturer/lifting-chain-61038.html) for things that [people](https://i.redd.it/fxq431q8hqe51.jpg) are expected to follow.

Reply:
This seems reasonable, but I am entirely unclear about what a callout link is, or how I go about creating one in markdown.

Can you give a concrete example?